virt-handler stops working on versions greater than v0.55.0 on RHEL 8.6 nodes with selinux enforcing

**What happened**:
We are running rke2 on RHEL8.6 with selinux set to enforcing.  After attempting to upgrade kubevirt, the virt-handler pods error out with the following virt-handler log failure:
```
...
"component":"virt-handler","level":"info","msg":"SELinux is reported as 'enforcing'","pos":"virt-handler.go:384","timestamp":"2022-09-29T18:18:24.591217Z"}
Error: error relabeling file /proc/self/fd/7 with label system_u:object_r:container_file_t:s0. Reason: operation not supported
Usage:
  virt-chroot selinux relabel [flags]

Examples:
virt-chroot selinux relabel <new-label> <file-path>

Flags:
  -h, --help          help for relabel
      --root string   safe root path which will be prepended to passed in files (default "/")

Global Flags:
      --cpu uint       cpu time in seconds for the process
      --memory uint    memory in bytes for the process
      --mount string   mount namespace to use
      --user string    switch to this targetUser to e.g. drop privileges

error relabeling file /proc/self/fd/7 with label system_u:object_r:container_file_t:s0. Reason: operation not supported
panic: error relabeling required files: error relabeling file root: /, relative: /dev/net/tun with label system_u:object_r:container_file_t:s0. Reason: exit status 1

goroutine 1 [running]:
main.(*virtHandlerApp).Run(0xc000306000)
	cmd/virt-handler/virt-handler.go:405 +0x2985
main.main()
	cmd/virt-handler/virt-handler.go:612 +0x54

```
After testing several versions, it looks like things work up to v0.55.0 and fail on v0.55.1 and above.  It looks like some changes were done to move to the golang.org.x/sys.unix for selinux labeling that might have broken things between those versions.  To verify, we set selinux to 'Permissive' on the kubernetes nodes and the virt-handler pods went into a 'running' state(but still displayed the error in the logs).

**What you expected to happen**:

virt-handler to start on versions greater than v0.55.0 on RHEL 8.6 based kubernetes nodes with selinux enabled.


**How to reproduce it (as minimally and precisely as possible)**:

Run kubernetes on RHEL 8.6 nodes with selinux enabled and run kubevirt greater than v0.55.0.

--- a/cmd/virt-chroot/BUILD.bazel
+++ b/cmd/virt-chroot/BUILD.bazel
@@ -17,7 +17,6 @@ go_library(
         "//vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs:go_default_library",
         "//vendor/github.com/opencontainers/runc/libcontainer/cgroups/fs2:go_default_library",
         "//vendor/github.com/opencontainers/runc/libcontainer/configs:go_default_library",
-        "//vendor/github.com/opencontainers/selinux/go-selinux:go_default_library",
         "//vendor/github.com/spf13/cobra:go_default_library",
         "//vendor/github.com/vishvananda/netlink:go_default_library",
         "//vendor/golang.org/x/sys/unix:go_default_library",

--- a/cmd/virt-chroot/selinux.go
+++ b/cmd/virt-chroot/selinux.go
@@ -7,7 +7,6 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/opencontainers/selinux/go-selinux"
 	"github.com/spf13/cobra"
 	"golang.org/x/sys/unix"
 
@@ -67,16 +66,17 @@ func RelabelCommand() *cobra.Command {
 
 			defer fd.Close()
 			filePath := fd.SafePath()
-			currentFileLabel, err := selinux.FileLabel(filePath)
-			if err != nil {
-				return fmt.Errorf("could not retrieve label of file %s. Reason: %v", filePath, err)
-			}
 
 			writeableFD, err := os.OpenFile(filePath, os.O_APPEND|unix.S_IWRITE, os.ModePerm)
 			if err != nil {
 				return fmt.Errorf("error reopening file %s to write label %s. Reason: %v", filePath, label, err)
 			}
 			defer writeableFD.Close()
+
+			currentFileLabel, err := getLabel(writeableFD)
+			if err != nil {
+				return fmt.Errorf("faild to get selinux label for file %v: %v", filePath, err)
+			}
 
 			if currentFileLabel != label {
 				if err := unix.Fsetxattr(int(writeableFD.Fd()), xattrNameSelinux, []byte(label), 0); err != nil {
@@ -89,4 +89,23 @@ func RelabelCommand() *cobra.Command {
 	}
 	relabelCommad.Flags().StringVar(&root, "root", "/", "safe root path which will be prepended to passed in files")
 	return relabelCommad
+}
+
+func getLabel(file *os.File) (string, error) {
+	// let's first find out the actual buffer size
+	var buffer []byte
+	labelLength, err := unix.Fgetxattr(int(file.Fd()), xattrNameSelinux, buffer)
+	if err != nil {
+		return "", fmt.Errorf("error reading fgetxattr: %v", err)
+	}
+	// now ask with the needed size
+	buffer = make([]byte, labelLength)
+	labelLength, err = unix.Fgetxattr(int(file.Fd()), xattrNameSelinux, buffer)
+	if err != nil {
+		return "", fmt.Errorf("error reading fgetxattr: %v", err)
+	}
+	if labelLength > 0 && buffer[labelLength-1] == '\x00' {
+		labelLength = labelLength - 1
+	}
+	return string(buffer[:labelLength]), nil
 }

